catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/CreateOperations.java

```diff
@@ -683,6 +697,43 @@ private CreateStorageRequest applyAttributeOverrides(CreateStorageRequest create
         createStorageRequest.getProperties()
                 .remove(Constants.ATTRIBUTE_OVERRIDES_KEY);
 
+        for (ContentItem contentItem : createStorageRequest.getContentItems()) {
+            Metacard metacard = metacardMap.get(contentItem.getId());
+            MetacardType metacardType = metacard.getMetacardType();
+
+            if (contentItem.getMetacard() != null) {
+                Metacard overrideMetacard = contentItem.getMetacard();
+                MetacardType overrideMetacardType = contentItem.getMetacard()
+                        .getMetacardType();
+
+                if (overrideMetacardType.equals(BasicTypes.BASIC_METACARD)) {
+                    overrideMetacardType.getAttributeDescriptors()
+                            .stream()
+                            .map(attributeDescriptor -> overrideMetacard.getAttribute(
+                                    attributeDescriptor.getName()))
+                            .filter(Objects::nonNull)
+                            .filter(attribute -> !attribute.getName()
+                                    .equals(Metacard.ID))
+                            .filter(attribute -> !attribute.getName()
+                                    .equals(Metacard.RESOURCE_URI))
+                            .forEach(metacard::setAttribute);
```

Comment:
I think you can rewrite this without side effects so that you do not need this if-else block and largely duplicate code.